### PR TITLE
22 Typedoc setup

### LINKED ISSUE #22: Typedoc setup
Setting up the typedoc documentation tool.

### PR DESCRIPTION
**Pull request by Felixsta123**
Getting automated documentation

### CODE REVIEW

**Comment by Felixsta123** [comment]
# Update
Typedoc is fully integrated.
Includes the dependencies, a fully commented **Button.tsx** as an example and an updated .gitignore.

## Usage
`npm run generate-docs`

File: tsconfig.json [hunk 1]
```diff
@@ -12,8 +12,25 @@
   "include": [
     "**/*.ts",
     "**/*.tsx",
+    "app/*",
```
Comment (PaulThblt) [inline tsconfig.json]: why this?

File: components/Button.tsx
Comment (PaulThblt) [inline components/Button.tsx:1]: This Button component is not using tailwind classes 👀

File: tsconfig.json (hunk 1 above)
Comment (Felixsta123) [reply tsconfig.json]: removed

File: components/Button.tsx
Comment (Felixsta123) [reply components/Button.tsx:1]: fixed

**Review by PaulThblt — APPROVED** [review]
Good fixes

**Pull request merged**